ui/src/app/routing/signed-in-app-routing.tsx

```diff
@@ -74,85 +62,84 @@ export const SignedInRoutes = React.memo(() => {
     </AppRoute>
-    <AppRoute path='/admin/workspace-audit'>
+    <AppRoute exact path='/admin/workspace-audit'>
       <WorkspaceAuditPage routeData={{title: 'Workspace Audit'}}/>
     </AppRoute>
-    <AppRoute path='/admin/workspace-audit/:workspaceNamespace'>
+    <AppRoute exact path='/admin/workspace-audit/:workspaceNamespace'>
       <WorkspaceAuditPage routeData={{title: 'Workspace Audit'}}/>
     </AppRoute>
-    <AppRoute path='/admin/workspaces/:workspaceNamespace/:nbName'>
+    <AppRoute exact path='/admin/workspaces/:workspaceNamespace/:nbName'>
       <AdminNotebookViewPage routeData={{pathElementForTitle: 'nbName', minimizeChrome: true}}/>
     </AppRoute>
-    <AppRoute path='/data-code-of-conduct'>
+    <AppRoute exact path='/data-code-of-conduct'>
       <DataUserCodeOfConductPage routeData={{title: 'Data User Code of Conduct', minimizeChrome: true}}/>
     </AppRoute>
-    <AppRoute path='/profile'>
+    <AppRoute exact path='/profile'>
       <ProfilePage routeData={{title: 'Profile'}}/>
     </AppRoute>
-    <AppRoute path='/nih-callback'>
+    <AppRoute exact path='/nih-callback'>
       <HomepagePage routeData={{title: 'Homepage'}}/>
     </AppRoute>
-    <AppRoute path='/ras-callback'>
+    <AppRoute exact path='/ras-callback'>
       <HomepagePage routeData={{title: 'Homepage'}}/>
     </AppRoute>
-    <ProtectedRoutes guards={[expiredGuard, registrationGuard]}>
-      <AppRoute path='/library'>
-        <WorkspaceLibraryPage routeData={{title: 'Workspace Library', minimizeChrome: false}}/>
-      </AppRoute>
-      <AppRoute path='/workspaces'>
-        <WorkspaceListPage routeData={{title: 'View Workspaces', breadcrumb: BreadcrumbType.Workspaces}}/>
-      </AppRoute>
-      <AppRoute path='/workspaces/build'>
-        <WorkspaceEditPage routeData={{title: 'Create Workspace'}} workspaceEditMode={WorkspaceEditMode.Create}/>
-      </AppRoute>
-      <AppRoute path='/workspaces/:ns/:wsid' exact={false}>
-        <WorkspaceWrapperPage intermediaryRoute={true} routeData={{}}/>
-      </AppRoute>
-    </ProtectedRoutes>
-  </React.Fragment>;
+    <AppRoute exact path='/library' guards={[expiredGuard, registrationGuard]}>
+      <WorkspaceLibraryPage routeData={{title: 'Workspace Library', minimizeChrome: false}}/>
+    </AppRoute>
+    <AppRoute exact path='/workspaces' guards={[expiredGuard, registrationGuard]}>
+      <WorkspaceListPage routeData={{title: 'View Workspaces', breadcrumb: BreadcrumbType.Workspaces}}/>
+    </AppRoute>
+    <AppRoute exact path='/workspaces/build' guards={[expiredGuard, registrationGuard]}>
+      <WorkspaceEditPage routeData={{title: 'Create Workspace'}} workspaceEditMode={WorkspaceEditMode.Create}/>
+    </AppRoute>
+    <AppRoute path='/workspaces/:ns/:wsid' exact={false} guards={[expiredGuard, registrationGuard]}>
+      <WorkspaceWrapperPage intermediaryRoute={false} routeData={{}}/>
+    </AppRoute>
+    <AppRoute exact path='*'>
```

Comment:
how does the not found path work if you're not signed in yet?

Reply:
It matches the non-exact `/` path for `SignedInPage` in `app-routing`, fails the `signInGuard`, and redirects to `/login`